### PR TITLE
fix diagflat fail if argument is not cupy.ndarray

### PR DESCRIPTION
**Pull request by kmaehashi**

### CODE REVIEW

**Comment by okuta** [comment]
Basically, I think that it is good for CuPy to request that the input is `cupy.ndarray`.
`cupy.diag` has unnecessary feature.

**Comment by kmaehashi** [comment]
I agree with that. > request that the input is `cupy.ndarray`
I guess the original code intends to accept scalar `v` for `xp.diag` and `xp.diagflat`.
So I fixed to check if input is scalar (use NumPy in this case) or CuPy.

File: cupy/creation/matrix.py [hunk 1]
```diff
@@ -22,17 +22,17 @@ def diag(v, k=0):
...
-            ret = cupy.zeros((size, size), dtype=v.dtype)
-            ret.diagonal(k)[:] = v
-            return ret
-        else:
-            return v.diagonal(k)
-    else:
+    if numpy.isscalar(v):
         return cupy.array(numpy.diag(v, k))
 
+    if v.ndim == 1:
```
Comment (okuta) [inline cupy/creation/matrix.py]: How about `if numpy.isscalar(v) or v.ndim == 1:` ?

File: cupy/creation/matrix.py [hunk 2]
```diff
@@ -45,10 +45,10 @@ def diagflat(v, k=0):
         cupy.ndarray: A 2-D diagonal array with the diagonal copied from ``v``.
 
     """
-    if isinstance(v, cupy.ndarray):
-        return cupy.diag(v.ravel(), k)
-    else:
-        return cupy.diag(numpy.ndarray(v).ravel(), k)
+    if numpy.isscalar(v):
+        return cupy.array(numpy.diagflat(v, k))
```
Comment (okuta) [inline cupy/creation/matrix.py]: How about `return cupy.diag(v, k)` ?

File: cupy/creation/matrix.py (hunk 1 above)
Comment (kmaehashi) [reply cupy/creation/matrix.py]: `v` may be a Python scalar (c.f., `np.isscalar(3)` returns `True`).

Comment (okuta) [reply cupy/creation/matrix.py]: OK.

**Comment by okuta** [comment]
LGTM!

**Pull request merged**